pkg/apiserver/diagnose/query.go

```diff
@@ -382,67 +524,154 @@ func (t totalValueAndTotalCountTableDef) genRow(values []string, subValues [][]s
 	}
 }
 
-func (t totalValueAndTotalCountTableDef) genSumarySQLs(startTime, endTime string, quantiles []float64) string {
-	sqls := []string{
-		fmt.Sprintf("select '%[1]s','' , sum(value) from metrics_schema.%[2]s where time >= '%[3]s' and time < '%[4]s'",
-			t.name, t.sumTbl, startTime, endTime),
-		fmt.Sprintf("select sum(value) from metrics_schema.%s where time >= '%s' and time < '%s'",
-			t.countTbl, startTime, endTime),
+type TotalValueAndTotalCount struct {
+	Name            string
+	Label           string
+	TotalSum        string
+	TotalCount      string
+	AggrQuantileMax string
+}
+
+func (t TotalValueAndTotalCount) toSlice() []string {
+	stringSlice := []string{
+		t.Name, t.Label, t.TotalSum, t.TotalCount,
 	}
-	for _, quantile := range quantiles {
-		sql := fmt.Sprintf("select max(value) as max_value from metrics_schema.%s where time >= '%s' and time < '%s' and quantile=%f",
-			t.tbl, startTime, endTime, quantile)
-		sqls = append(sqls, sql)
-	}
-	fields := ""
-	tbls := ""
-	for i, sql := range sqls {
-		if i > 0 {
-			fields += ","
-			tbls += "join "
+	if len(t.AggrQuantileMax) == 0 {
+		return stringSlice
+	}
+	stringSlice = append(stringSlice, strings.Fields(t.AggrQuantileMax)...)
+	return stringSlice
+}
+
+func (t totalValueAndTotalCountTableDef) querySummary(db *gorm.DB, startTime, endTime string, quantiles []float64) ([]string, error) {
+	tables := fmt.Sprintf("metrics_schema.%s t0", t.sumTbl)
+	tables += fmt.Sprintf(", metrics_schema.%s t1", t.countTbl)
+
+	aggrQuantileMaxField := ""
+	tableNum := 2
+	for range quantiles {
+		tableAlias := fmt.Sprintf("t%v", tableNum)
+		tables += fmt.Sprintf(", metrics_schema.%s %s", t.tbl, tableAlias)
+		if tableNum == 2 {
+			aggrQuantileMaxField += ", "
+		} else {
+			aggrQuantileMaxField += "||' '||"
 		}
-		fields += fmt.Sprintf("t%v.*", i)
-		tbls += fmt.Sprintf(" (%s) as t%v ", sql, i)
+		aggrQuantileMaxField += fmt.Sprintf("max(%v.value)", tableAlias)
+		tableNum++
 	}
-	joinSQL := fmt.Sprintf("select %v from %v", fields, tbls)
-	return joinSQL
+	if len(aggrQuantileMaxField) == 0 {
+		aggrQuantileMaxField = "''"
+	}
```

Comment:
you can avoid building the field list if `quantiles []float64` are changed to a closed set, like this:

```golang
type QuantileFields {
  Q999 float64
  Q99 float64
  Q90 float64
  Q80 float64
}

type Quantiles {
  Q999 bool
  Q99 bool
  Q90 bool
  Q80 bool
}
```

When using it:

```golang
type XxxRow {
  QuantileFields
}

func (...) querySummary(..., quantiles: Quantiles) ([]XxxRow, error) {
  ...
}
```

Actually quantiles are indeed a closed set:

```golang
quantiles: []float64{0.999, 0.99, 0.90, 0.80},
```